test/timeouts.test.js

```diff
@@ -0,0 +1,153 @@
+describe('Test timeouts', function () {
+  var async = require('async');
+  var MockRedisClient = require('./mockRedisClient.js');
+  var proxyquire = require('proxyquire');
+
+  // Reset before each test
+  var ShardedRedis;
+
+  /********************************
+   * Setup
+   ********************************/
+
+  beforeEach(function () {
+    // Get sharded-redis-client with the redis clients stubbed out
+    ShardedRedis = proxyquire('../sharded-redis-client.js', {
+      redis: MockRedisClient
+    });
+    jasmine.clock().install();
+  });
+
+  afterEach(function () {
+    jasmine.clock().uninstall();
+  });
+
+  /********************************
+   * Tests
+   ********************************/
+
+  it('should cb with an error on a read if redis is not responding if read timeouts enabled', function (done) {
+    spyOn(MockRedisClient.prototype, 'get');
+    var numMasterHosts = 1;
+    var options = {
+      readTimeout: 500
+    };
+    var key = 'key';
+    var redisHosts = generateRedisHosts(numMasterHosts);
+
+    var shardedClient = new ShardedRedis(redisHosts, options);
+    shardedClient.get(key, function (err) {
+      expect(err).toBeTruthy();
+      expect(MockRedisClient.prototype.get).toHaveBeenCalledTimes(1);
+      expect(err.message).toBe('Redis call timed out');
```

Comment:
I mentioned this above, but usually when I test for error messages I also expose them.

`expect(err.message).toBe(client.errors.TIMEOUT);`

To make the test less brittle and to make them available to clients.